delta_backend/src/ConversionChecker.py

```diff
@@ -85,25 +93,120 @@ def convertData(self, expressionType, expressionRule, fieldName, fieldValue):
                     expressionRule, fieldName, fieldValue, self.summarise, self.report_unexpected_exception
                 )
             case _:
-                return "Schema expression not found! Check your expression type : " + expressionType
+                raise ValueError("Schema expression not found! Check your expression type : " + expressionType) 
+
+    # Utility function for logging errors
+    def _log_error(self, fieldName, fieldValue, e, code=ExceptionMessages.RECORD_CHECK_FAILED):
+        if isinstance(e, Exception):
+            message = ExceptionMessages.MESSAGES[ExceptionMessages.UNEXPECTED_EXCEPTION] % (e.__class__.__name__, str(e))
+        else:
+            message = str(e)  # if a simple string message was passed 
+
+        # Ensure message-level deduplication
+        if any(existing.get("message") == message for existing in self.errorRecords):
```

Comment:
Does this mean if we have the same type of error for two different fields, then we'll only report one of the errors? Maybe we should deduplicate on the entire error object, or not deduplicate at all?

Reply:
How would we actually end up with duplicate error messages?